[MIG] account_bank_statement_import_paypal: Migration to 12.0

cubells opened the pull request:
Standard migration

cc @Tecnativa

Commit:
Add module account_bank_statement_import_paypal
Fix spelling mistake
Add methods to inherit for language-specific or country-specific stuff Port to new API
FIX super
Update travis config

Commit:
Consider refund line
Avoid to return paypal_email_account as a bank account
Put file transaction ref in name to work with base_transaction_ref
Remove thousands separator

mourad-ehm committed:
adapt code to the new paypal format
error when several line defined in currency != company currency
handel utf-8 file
handel csv with quotechar "
skip BOM character in utf-8

sebastienbeau committed:
refactor paypal import in order to use new bank statement from Paypal
Remove deprecated information in readme
solve issue on currency the main currency is not the first line (line are ordered by currency name). So we should take the journal currency or company currency to know the right currency to use. Also add full header to avoid issue if the paypal header change

cubells committed:
[MIG]account_bank_statement_import_paypal: Migration to 11.0

oca-travis committed:
[UPD] Update account_bank_statement_import_paypal.pot

sbidoul committed:
[FIX] title in readme fragment

OCA-git-bot committed:
[UPD] README.rst

cubells committed:
[IMP] account_bank_statement_import_paypal: create paypal maps

cubells committed:
[IMP] account_bank_statement_import_paypal: create paypal maps

oca-travis committed:
[UPD] Update account_bank_statement_import_paypal.pot

pedrobaeza commented on account_bank_statement_import_paypal/data/paypal_map_data.xml:
```diff
@@ -0,0 +1,114 @@
+<odoo noupdate="1">
+
+    <record id="paypal_map" model="account.bank.statement.import.paypal.map">
+        <field name="name">Paypal Monthly Statement</field>
```
Add ` (English)` at the end for recognizing that is in that language

pedrobaeza reviewed (changes requested):
I'm missing here Spanish paypal map configuration, and check also runbot warnings:
```
2019-04-24 10:37:08,482 161 WARNING openerp_test odoo.models: The model account.bank.statement.import.paypal.map has no _description
2019-04-24 10:37:08,483 161 WARNING openerp_test odoo.models: The model account.bank.statement.import.paypal.map.line has no _description
2019-04-24 10:37:08,483 161 WARNING openerp_test odoo.models: The model wizard.paypal.map.create has no _description
```

MiquelRForgeFlow reviewed (changes requested):
I would like to see separated the IMP commit of v11 and the MIG commit to v12. Now they are fused :S

cubells commented:
> I'm missing here Spanish paypal map configuration, and ...

Do you want I add an spanish paypal map in data?

pedrobaeza commented:
> Do you want I add an spanish paypal map in data?

Yes, as it's needed for our customer, and it's the same valid predefined data as English one.

cubells commented:
This is WIP please.

gurneyalex committed:
[FIX] float parsing in paypal import

In English version of the files, paypal uses ',' as thousands separator and '.' as
decimal separator -> the previous code would populate the bank statement with
wrong values without complaining.

We fix this by allowing to configure the separators in the paypal map.

pedrobaeza commented:
@victormmtorres please re-take this and include #211 here

Commit:
[MIG] account_bank_statement_import_paypal: Migration to 12.0

victor-ditay commented:
cc @pedrobaeza changes done

MiquelRForgeFlow commented on account_bank_statement_import_paypal/models/account_bank_statement_import_paypal_map.py:
```diff
@@ -0,0 +1,108 @@
+
+class AccountBankStatementImportPaypalMap(models.Model):
+    _name = 'account.bank.statement.import.paypal.map'
+
```
add `_description`

MiquelRForgeFlow commented on account_bank_statement_import_paypal/models/account_bank_statement_import_paypal_map.py:
```diff
@@ -0,0 +1,108 @@
+class AccountBankStatementImportPaypalMapLIne(models.Model):
+    _name = 'account.bank.statement.import.paypal.map.line'
+    _order = "sequence asc, id asc"
+
```
add `_description`

MiquelRForgeFlow commented on account_bank_statement_import_paypal/wizards/create_map_lines_from_file.py:
```diff
@@ -0,0 +1,38 @@
+
+class WizardPaypalMapCreate(models.TransientModel):
+    _name = 'wizard.paypal.map.create'
+
```
add `_description`

MiquelRForgeFlow commented on account_bank_statement_import_paypal/models/account_bank_statement_import_paypal_map.py:
```diff
@@ -0,0 +1,108 @@
+                separators[self.float_decimal_sep])
+
+
+class AccountBankStatementImportPaypalMapLIne(models.Model):
```
...LIne -> ...Line

victor-ditay commented:
Changes done @mreficent

MiquelRForgeFlow commented:
... and runbot is being stupid.

MiquelRForgeFlow reviewed (approved):
LGTM
